### ISSUE TITLE
Missing accent and special character support

### ISSUE THREAD

**Issue by ed359** [valentjn/languagetool-languageserver#56]
**Describe the bug**
Several accents and special characters are missing support. I am particularly interested in proper handling of \H{o} -> ő. See the details below.

**Steps to reproduce**

1.  Save the latex code given below as a .tex file
2. Open it in VSCode with the LTeX plugin enabled
3. Observe the poor handling of some accents. One can add Erdős to the dictionary but still get a spelling error on Erd\H{o}s, and the Polish words are marked as errors when the latex command is used to insert the accent (rather than using the correct unicode letter).

**Expected behavior**
The accent commands \H, \k and similar should be correctly parsed and interpreted by LTeX

**Sample document**

<details>

```
\documentclass{article}

\usepackage[utf8]{inputenc} % not necessary with newer versions of latex
\usepackage[T1]{fontenc} % necessary to allow the various accents
\usepackage{lmodern} % (optional) load a modern font
\usepackage{hyperref} % for hyperlinks

\begin{document}

\section{Information on diacritics in latex}

\url{https://tex.stackexchange.com/tags/accents/info}\\
\url{https://en.wikibooks.org/wiki/LaTeX/Special_Characters}

\section{Missing support in LTeX}


\begin{enumerate}
    \item Erd\H{o}s should be recognised as Erdős
% LTeX: SETTINGS language=pl-PL
    \item miesi\k{a}c should be recognised as miesiąc 
    \item miesi\k{e}cznie should be recognised as miesięcznie
    \item b\l\k{a}d should be recognised as błąd
\end{enumerate}

% LTeX: SETTINGS language=en-GB

\subsection{Further unsupported diacritics (without examples)}
\begin{enumerate}
    \item o with a bar under the letter: \b{o}
    \item u with a dot under the letter: \d{u}
    \item o with a breve over the letter: \u{o}
    \item s, c with a caron/h\'{a}\v{c}ek over the letter: \v{s}, \v{c}
    \item tie accents \t{oo} or \t{o}o
    \item dotless i, j: \i, \j
\end{enumerate}

\section{Diacritics with support}

\begin{enumerate}
    \item \AA, \O, \aa, \ss, \o: see \href{https://github.com/valentjn/ltex-ls/blob/1b6d0e8fb4753d43ffd968f4ac87b38d6a61eb36/ltexls-core/src/main/java/org/bsplines/ltexls/parsing/latex/LatexAnnotatedTextBuilder.java#L563}{this url}.
    \item Grave accent on AEIOUaeiou, acute accent on AEIOUYaeiouy, circumflex on AEIOUYaeiouy, tilde on AEINOUaeinou, umlaut on AEIOUYaeiouy, macron on AEIOUYaeiouy, dot on AEIOaeo, cedilla on Cc, ring on AUau,
see \href{https://github.com/valentjn/ltex-ls/blob/1b6d0e8fb4753d43ffd968f4ac87b38d6a61eb36/ltexls-core/src/main/java/org/bsplines/ltexls/parsing/latex/LatexAnnotatedTextBuilder.java#L290}{this url}.
\end{enumerate}

\end{document}
```

</details>

**LTeX configuration**
N/A, no configuration necessary to observe bug

**LTeX LS log**
N/A, the bug produces no log output

**Version information**
List here the version information of the relevant software.

- Operating system: Windows 10
- ltex-ls: 10.0.0
- Java: unknown (this is irrelevant to the bug)

**Additional context/information**
None

**Comment by ed359** [comment]
This PR should fix the issues I've been having: https://github.com/valentjn/ltex-ls/pull/57

**Comment by valentjn** [comment]
Changing this to feature request as there is no way LT<sub>E</sub>X can cover all of Unicode, so these accents were previously out of feature scope.

**Comment by valentjn** [comment]
Fix released in 11.0.0 (corresponds to vscode-ltex 10.0.0).
